### PR TITLE
Conditional operator suffixes for property names when matching

### PR DESCRIPTION
**Pull request by jonathanbaugh**
Hello!

I'm just beginning to familiarize myself with graph databases and I was looking for a module that lets me interact with it more "Pythonically." py2neo looks great!

This pull request enables additional property comparisons by appending operators to arguments in the match statements. This is found in other ORMs and I thought it'd be neat in py2neo :-)

```
# Find people born after 1985
graph.match("Person").where("born > 1985")
graph.match("Person").where(("born > {1}", {"1": 1985}))
graph.match("Person", born__gt=1985)  # Pull request adds this
```

Operators include:
`__gt`: Greater than
`__gte`: Greater than or equal to
`__lt`: Less than
`__lte`: Less than or equal to
`__exact`: Explicit equals (`=`)
`__not`: Not equal to (`<>`)
`__startswith` and `__endswith`: Starts or ends with (`STARTS WITH` or `ENDS WITH`)
`__contains`: Contains operator (`CONTAINS`)

### CODE REVIEW

File: docs/matching.rst [hunk 1]
```diff
@@ -22,6 +22,39 @@ For a simple match by label and property::
         >>> matcher.match("Person", name="Keanu Reeves").first()
         (_224:Person {born:1964,name:"Keanu Reeves"})
 
+Many types of comparisons can be made with operator suffixes. By appending the opeartor suffix you change the operator used in Cypher expression.
+
+        +--------------------------+--------------+-------------+------------------------------------------------------------+
+        | Description              | Suffix       | Operator    | Example                                                    |
+        +--------------------------+--------------+-------------+------------------------------------------------------------+
+        | Explicit Equal           | __exact      | =           | >>>matcher.match("Person", name__exact="Kevin Bacon")         |
```
Comment (technige) [inline docs/matching.rst]: Looks like an unintended `>>>` was left here.

File: docs/matching.rst [hunk 2]
```diff
@@ -22,6 +22,39 @@ For a simple match by label and property::
...
+        |                          |              |             | MATCH (_:Person) WHERE name = "Kevin Bacon" RETURN _       |
+        +--------------------------+--------------+-------------+------------------------------------------------------------+
+        | Not Equal                | __not        | <>          | matcher.match("Person", name__not="Rick Astley")           |
```
Comment (technige) [inline docs/matching.rst]: Bonus points for Rick Astley. Although some of his new stuff has been pretty good :)

File: test/integration/test_matching.py [hunk 3]
```diff
@@ -83,6 +83,69 @@ def test_custom_conditions_with_parameters(self):
         assert first["name"] == "Keanu Reeves"
         assert first["born"] == 1964
 
+    def test_special_parameters_gt(self):
+        year = 1985
+        found = list(self.matcher.match("Person", born__gt=year))
+        assert len(found) >= 1
```
Comment (technige) [inline test/integration/test_matching.py]: Minor point: this could just be `assert found`.

**Comment by technige** [comment]
Hi @jonathanbaugh! Thanks for this PR, this is a _really_ neat addition to the OGM, and a high quality PR. I've added a couple of small comments, nothing major.

Where does the double underscore notation come from? I can't think of anything better, I was just wondering if this had precedent elsewhere.

**Comment by jonathanbaugh** [comment]
Hey @technige,

Thanks for the feedback! I've updated the PR to reflect the changes.

I first came across the double underscore notation in Django. I swear I've seen it other places but can't find it now :-P

Thanks,
Jonathan

**Pull request merged**